dev/image-guidelines.md

```diff
@@ -0,0 +1,335 @@
+
+For line thickness ("**stroke width**") we use "**extra bold**" for annotation arrows, but this is subjective choice, since line thickness is ultimately relative to the dimensions of the rendered image.
+
+
+![](image-guidelines/excalidraw_defaults.png)
+
+
+
+## Line diagrams
+
+We also use Excalidraw for line diagrams. These can be drawn from scratch but Excalidraw also allows importing some Mermaid markdown which can be useful for quickly creating initial diagram structures.
+
+The following image highlights the recommended Excalidraw settings for our diagrams. Generally these are similar to the settings for annotations but note the following differences:
+
+- stroke width is **Bold** (not Extra Bold)
+- stroke color is black (not blue)
+
+![](image-guidelines/line_diagram_settings.png)
+
+The diagram heading is made bold by choosing our #2 font which is the same font as our #1 font but a bold variant. This is necessary because Excalidraw doesn't have any options for bolding or italicizing fonts.
+
+## Exporting images
+
+### Formats and settings
+
+![](image-guidelines/excalidraw_export_settings.png)
+
+We export our images to SVG format, and not PNG.
+
+When exporting images from Excalidraw we include "Background" which ensures that the background is not transparent (i.e. an opaque white background) plus a small margin around it.
+
+For SVG exports the Scale setting is irrelevant to quality: the embedded screenshot will be base64-encoded at it's original size, even if you've scaled the screenshot on the Excalidraw canvas. The image dimensions are related to the vector layer and if this is resized by the web server then the embedded raster image will be scaled to match.
+
+You should resize the screenshot so that it's comfortable to work with on the canvas at 100% zoom. When exported at 1x scale the SVG will be sized at the dimensions of a bounding box around the elements on your canvas. The line thicknesses and font sizes are relative to your canvas.
+
+![](image-guidelines/embedded-image-scaling.png)
+
+
+Although it is possible to edit SVG images externally they cannot be imported back into Excalidraw.
+So for annotated screenshots and block diagrams we save the `.excalidraw` file alongside the exported SVG or PNG image, using the same file basename.
+This allows others to easily make changes in future.
+If the image is reused in multiple places on the docs website (or other Velociraptor repos such as presentations) we only need to keep *one instance* of the `.excalidraw` file.
+
+
+> [!NOTE]
+> Regardless of the export format, remember to also save a copy of the drawing in Excalidraw format so that others can edit it in future, if necessary.
+>
+
+Annotated screenshots exported from Excalidraw as SVG images contain a base64-encoded copy of the screenshot. The saved `.excalidraw` file does too. So it is not necessary to keep a copy of the original screenshot PNG.
+
+### Fonts in SVG images
```

Comment:
This section should just say - fonts are normalized to the standard fonts used throughout the site so dont worry about specific fonts in your drawings.

The technical details are not relevant and could change in future - also not really actionable by readers.

Reply:
Sure once we've settled on a process that works and the specifics of that comes down to these choices:

1. Use Excalidraw.com hosted app. Use their fonts, use carefully designed bounding boxes and alignments, and accept that quite often the SVGs could be significantly messed up after the font swap, something which we'll only know about once the images have been published to the docs site. Also expect breakages without warning as the online app evolves.

2. Use a locally installed instance of the official Excalidraw. That eliminates the unexpected breakages but otherwise has all the same downsides as #1.

3. Use a locally installed version of the excalidraw-zh fork, which allows WYSIWYG editing and generates SVGs with a 1-to-1 font spec that can be swapped without unexpected consequences. It provides all the desired functionality that the official version offers, or even promises to offer in future. It's works and it's not going to suddenly break.

Reply:
No problem - it is actually not that bad because usually when writing docs we have to run hugo anyway so the font substitution happens in real time and you can see updates to the svg immediately.

Reply:
Ok, we still need to pick one though because they all embed different font names in the vector part and we just want to replace the font definition part to achieve the right mapping.

Reply:
Just added a commit that shows the font defs swap we need to make when exporting SVG from excalidraw.com